util/intern.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+namespace atlas {
+namespace util {
+
+class StrRef {
+ public:
+  StrRef() = default;
+  bool operator==(const StrRef& rhs) const { return data == rhs.data; }
+  const char* data = nullptr;
+};
+
+const StrRef& intern_str(const char* string);
+const StrRef& intern_str(const std::string& string);
+const char* to_string(const StrRef& ref);
```

Comment:
Why not just have a `get()` member on `StrRef` similar to a java `AtomicReference`?

Reply:
Mainly because on some implementations I was playing with you needed to have access to the pool to get the string value. I like your suggestion of `get()` instead of `to_string`/`to_cstring`